core/src/solution_submission/retry.rs

```diff
@@ -6,9 +6,60 @@ use ethcontract::{
     errors::{ExecutionError, MethodError},
     U256,
 };
+use futures::future::{BoxFuture, FutureExt as _};
 
 use super::MIN_GAS_PRICE_INCREASE_FACTOR;
 
+pub struct Args<'a> {
+    pub contract: &'a (dyn StableXContract + Sync),
+    pub batch_index: u32,
+    pub solution: Solution,
+    pub claimed_objective_value: U256,
+    pub gas_price_estimating: &'a (dyn GasPriceEstimating + Sync),
+    pub gas_price_cap: U256,
+    pub nonce: U256,
+}
+
+pub trait RetryingWithGasPriceIncrease {
+    fn retry<'a>(&self, args: Args<'a>) -> BoxFuture<'a, Result<(), MethodError>>;
+}
+
+pub struct RetryWithGasPriceIncrease {}
+
+impl RetryingWithGasPriceIncrease for RetryWithGasPriceIncrease {
+    fn retry<'a>(&self, args: Args<'a>) -> BoxFuture<'a, Result<(), MethodError>> {
+        retry(
```

Comment:
Any reason this doesn't take an `Args` parameter? If the reason is to not change the `retry` method than cant you just destructure the args parameter?

For example this is valid Rust:
```
struct Foo {
    bar:i32,
    baz:i64,
}
fn foo(Foo { bar, baz }: Foo) {
    dbg!((bar, baz));
}
```